src/parquet/arrow/schema.cc

```diff
@@ -485,14 +485,19 @@ Status FieldToNode(const std::shared_ptr<Field>& field,
     case ArrowType::TIMESTAMP: {
       auto timestamp_type = static_cast<::arrow::TimestampType*>(field->type().get());
       auto unit = timestamp_type->unit();
-      type = ParquetType::INT64;
       if (unit == ::arrow::TimeUnit::MILLI) {
+        type = ParquetType::INT64;
         logical_type = LogicalType::TIMESTAMP_MILLIS;
       } else if (unit == ::arrow::TimeUnit::MICRO) {
+        type = ParquetType::INT64;
         logical_type = LogicalType::TIMESTAMP_MICROS;
+      } else if (unit == ::arrow::TimeUnit::NANO) {
```

Comment:
As weird as a it sounds, I would like to have a flag that needs to be set to true to write int96 timestamps, e.g. `support_deprecated_int96_timestamps`. Otherwise e.g. timestamps coming from Pandas will always be saved to Int96. That would be quite destructive in the deprecation of Int96.